File: src/main/java/com/example/oauth/config/properties/OAuth1ClientProperties.java
package com.example.oauth.config.properties;

import com.example.oauth.model.oauth.OAuthProvider;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class OAuth1ClientProperties {
    private OAuthProvider provider;
    private String clientId;
    private String clientSecret;
    private String callbackUri;
    private String userInfoUri;
    private String userNameAttribute;

    public OAuth1ClientProperties(OAuthProvider provider) {
        this.provider = provider;
    }
}

File: src/main/java/com/example/oauth/config/properties/OAuth1ClientProperties.kt
package com.example.oauth.config.properties

import com.example.oauth.model.oauth.OAuthProvider

class OAuth1ClientProperties(
    val provider: OAuthProvider,
    var clientId: String = "",
    var clientSecret: String = "",
    var callbackUri: String = "",
    var userInfoUri: String = "",
    var userNameAttribute: String = "",
)